src/esw/cameras.py

```diff
@@ -1,387 +1,232 @@
-            return True
-        elif not is_streaming and self.video_source is None:
-            return False
-        else:
-            assert False, "Should not have entered invalid state in is_streaming()."
+class StreamManager:
 
-
-class StreamingManager:
-    # _services is a list of size 2 representing the requested cameras from the primary and secondary laptops.
-    # For each laptop, there is a list of size 4 that represents information on the requested device and its resolution.
-    _services: List[List[CameraCmd]]
-
-    _device_lock: threading.Lock
-
-    # _list_of_resolution_options is a list of sizes that represents the different options for resolutions
-    # (0 for low, 2 for high).
-    # For each resolution, there is a Resolution object that has an option number and a list of strings that is passed
-    # into the jetson.utils function that is needed to create that specific resolution.
-    _list_of_resolution_options: List[Resolution]
-
-    # _endpoints is a list of size 8, where the first four strings represent the endpoints of the primary laptop
-    # and the last four strings represent the endpoints of the secondary laptop.
-    # Endpoints are just a combination of an IP and a port.
-    _endpoints: List[str]
-
-    # _service_streams_by_endpoints is a dictionary that maps each endpoint to the device that is being streamed to it.
-    _service_streams_by_endpoints: Dict[str, Tuple[int, int]]
-
-    # _video_devices is a list of jetson.utils objects. It can be None.
-    _video_devices: List[VideoDevices]
-    _active_devices: int
-
-    # determined by hardware. this is a constant. do not change.
-    MAX_CAMERA_DEVICES: int = 4
+    NUM_DIFFERENT_IPS = 2
+    MAX_NUM_POSSIBLE_CAMERAS = 4
 
     def __init__(self):
 
-        self._services = [
-            [CameraCmd(-1, 0), CameraCmd(-1, 0), CameraCmd(-1, 0), CameraCmd(-1, 0)],
-            [CameraCmd(-1, 0), CameraCmd(-1, 0), CameraCmd(-1, 0), CameraCmd(-1, 0)],
-        ]
-        self._video_devices = []
-        for i in range(rospy.get_param("cameras/max_video_device_id_number")):
-            skip_every_other_device = rospy.get_param("cameras/skip_every_other_device")
-            if skip_every_other_device:
-                self._video_devices.append(VideoDevices(i * 2))
-            else:
-                self._video_devices.append(VideoDevices(i))
-        self._endpoints = [
-            rospy.get_param("cameras/endpoints/primary_0"),
-            rospy.get_param("cameras/endpoints/primary_1"),
-            rospy.get_param("cameras/endpoints/primary_2"),
-            rospy.get_param("cameras/endpoints/primary_3"),
-            rospy.get_param("cameras/endpoints/secondary_0"),
-            rospy.get_param("cameras/endpoints/secondary_1"),
-            rospy.get_param("cameras/endpoints/secondary_2"),
-            rospy.get_param("cameras/endpoints/secondary_3"),
+        self.stream_process_list: List[List[Union[Process, Any]]] = [
```

Comment:
You could type hint at the top of the class if you want to, though I guess here works just as well. More importantly, what's the point of Union[Process, Any] instead of just Any

Reply:
So at first I was trying to do Union[Process, None], but I still faced issues... so then I did Union[Process, Any] without thinking too hard lol. SO yes we should change to Any